packages/vitest/src/runtime/workers/base.ts

```diff
@@ -0,0 +1,47 @@
+import { ModuleCacheMap } from 'vite-node/client'
+import type { WorkerGlobalState } from '../../types/worker'
+import { provideWorkerState } from '../../utils/global'
+import type { ContextExecutorOptions, VitestExecutor } from '../execute'
+import { getDefaultRequestStubs, startVitestExecutor } from '../execute'
+import type { MockMap } from '../../types/mocker'
+
+let _viteNode: VitestExecutor
+
+const moduleCache = new ModuleCacheMap()
+const mockMap: MockMap = new Map()
+
+async function startViteNode(options: ContextExecutorOptions) {
+  if (_viteNode)
+    return _viteNode
+
+  _viteNode = await startVitestExecutor(options)
+  return _viteNode
+}
+
+export async function runBaseTests(state: WorkerGlobalState) {
+  const { ctx } = state
+  // state has new context, but we want to reuse existing ones
+  state.moduleCache = moduleCache
+  state.mockMap = mockMap
+
+  provideWorkerState(globalThis, state)
+
+  if (ctx.invalidates) {
+    ctx.invalidates.forEach((fsPath) => {
+      moduleCache.delete(fsPath)
+      moduleCache.delete(`mock:${fsPath}`)
+    })
+  }
+  ctx.files.forEach(i => state.moduleCache.delete(i))
+
+  const [executor, { run }] = await Promise.all([
+    startViteNode({ state, requestStubs: getDefaultRequestStubs() }),
+    import('../runBaseTests'),
```

Comment:
Why dynamic import? `runBaseTests` is not defined as separate Rollup entry, like `runVmTests` is. Should it?
I don't see any side effects on `runBaseTests`, so if it's not meant to be in a separate bundle I guess it could be imported on top level?

Reply:
It imports from Vitest which relies on __vitest_worker__ which is not defined until this function is called